Added necessary bootstrap files

DilushaLakshan opened the pull request:
After resolving the conflict, necessary bootstrap files were lost. They were added.

DilushaLakshan committed:
Delete bootstrap.bundle.js

DilushaLakshan committed:
Delete bootstrap.css

DilushaLakshan committed:
Delete connection.php

DilushaLakshan committed:
Delete index.php

DilushaLakshan committed:
Delete script.js

DilushaLakshan committed:
Delete styles.css

DilushaLakshan committed:
Convert hardcode data into fetch from the database

DilushaLakshan committed:
Completed update tour package and modified tour planning process and add-destination process for storing the packing list ids.

DilushaLakshan committed:
Mainly completed the following modules.
Tour package filtering
Send emails when signing in
Bookings and booking confirmation

Modify the following modules
Add tour package
Update tour packages
Add destinations
Update destinations
Add vehicle
Update vehicles

Also implement the footer and a part of visual dashboard using static data and Chart.js library.

DilushaLakshan committed:
Mainly completed the following modules.
Tour package filtering
Send emails when signing in
Bookings and booking confirmation

Modify the following modules
Add tour package
Update tour packages
Add destinations
Update destinations
Add vehicle
Update vehicles

Also implement the footer and a part of visual dashboard using static data and Chart.js library.

DilushaLakshan committed:
Concluded merge and resolved conflicts

DilushaLakshan committed:
add bootstrap.bundle.js file and bootstrap.css file

DilushaLakshan committed:
Merge branch 'development' of https://github.com/DilushaLakshan/RoamRadienceTravelers into development

DilushaLakshan committed:
update missing bootstrap files

DilushaLakshan committed:
Completed the manage hotel process and resolve the problems occured in following modules. Bookings
Manage vehicles

dulanjayabhanu commented on script.js:
```diff
@@ -193,6 +193,26 @@ function updateStaffMember(mID) {
 
+// block staff user
+function blockUnblockUser(mID, statusID) {
+    var mID = mID;
```
This variable assign operation is useless. You can directly use mID, statusID local variables.

dulanjayabhanu commented on script.js:
```diff
@@ -264,13 +288,13 @@ function sendDestinationDetails() {
 function updateDestination(desID) {
     var desID = desID;
```
This variable assignment operation is useless. You can directly use desID local variable.

dulanjayabhanu commented on script.js:
```diff
@@ -382,6 +406,47 @@ function addHotel() {
 
+// update hotel details
+function updateHotel(id) {
+    var hID = id;
```
You can directly use `id` variable

dulanjayabhanu commented on script.js:
```diff
@@ -427,10 +492,10 @@ function addVehicle() {
 // update vehicle details
 function updateVehicle(vID) {
     var vID = vID;
```
You can directly use `vID` variable.

dulanjayabhanu commented on script.js:
```diff
@@ -779,8 +844,9 @@ function sendBookingData(packageID) {
+function updateStatus(travelerID, bookingID, statusID) {
     var travelerID = travelerID;
+    var bookingID = bookingID;
     var statusID = statusID;
```
This variable assignment operation is useless. You can directly usetravelerID, bookingID, and statusID variables.

dulanjayabhanu commented on update-destination.php:
```diff
@@ -10,6 +10,7 @@
     <title>Update Destnation</title>
     <link rel="stylesheet" href="bootstrap.css">
     <link rel="stylesheet" href="styles.css">
+    <script src="https://cdn.ckeditor.com/4.16.2/standard/ckeditor.js"></script>
```
Don't link the script files in the header section. While it may technically work, the standard approach is to place the script link just before the body closing tag.

DilushaLakshan replied:
got it.

dulanjayabhanu reviewed (approved):
It's fine to go ahead with the merge. There are a few minor bugs, fix it.

DilushaLakshan committed:
Change the UI/UX of front-end and back-end.
Included Sweet Alert API for success, warning and error messages instead of alert.
Implemented a part of the rating system.
Implemented part of the promotion part.

DilushaLakshan committed:
Fixed some bugs in manage promotions functionality and fixed issues in the navbar

dulanjayabhanu commented on add-destination.php:
```diff
@@ -12,156 +15,165 @@
                                                     </div>
                                                     <div class="col-10">
-                                                        <p><?php echo $categoryData["name"]; ?></p>
+                                                        <label for=""><i>No categories available</i></label>
```
Use the `span` tag. The label tag is used to represent some input fields.

dulanjayabhanu commented on add-hotel.php:
```diff
@@ -10,119 +13,129 @@
-                                <input type="text" class="w-100" id="h-name">
-                            </div>
+    <?php
+    if (isset($_SESSION["user"])) {
```
Web content is only visible when the `user session object` is link with user. however, if the user session is lost, a **blank web page** may be rendered. So, include the `else` code fragment and redirect to the proper page when the user session object is missing.

dulanjayabhanu commented on add-promotion.php:
```diff
@@ -0,0 +1,153 @@
+            </div>
+        </div>
+    <?php
+    }
```
Include the `else` code fragment and redirect to the proper page when the user session object is missing.

dulanjayabhanu commented on add-tourpackage.php:
```diff
@@ -178,158 +160,240 @@
+                                                ?>
                                                 <div class="col-12 col-md-6 col-lg-6">
-                                                    <input type="checkbox" name="services" value="<?php echo $serviceData['id']; ?>"> <?php echo $serviceData["name"]; ?>
+                                                    <input type="checkbox" name="type" value="0"> no Data
```
Remove the input field. The message is enough

dulanjayabhanu commented on add-tourpackage.php:
```diff
@@ -178,158 +160,240 @@
+                                            <input class="form-control w-100" type="file" id="main-image" accept=".png, .jpg, .jpeg" onclick="mainImagePreview();">
+                                        </div>
+                                        <div class="col-12 mt-2">
+                                            <center><img alt="" id="m-image" style="width: 300px; height: 400px; object-fit: cover;" class="img-fluid rounded-2" src="resources/images/default -image.svg"></center>
```
Try wrapping all this style in a class

dulanjayabhanu commented on add-tourpackage.php:
```diff
@@ -178,158 +160,240 @@
-                                <button class="btn sbt-button" onclick="addTourPackage();">ADD</button>
+                        </div>
+
+                        <!-- destination selecttion model -->
```
Do not create 3 models for each process. Create just one model and put all 3 design in it. Then you can manage this all 3 UIs using JavaScript.

dulanjayabhanu commented on addPromotionProcess.php:
```diff
@@ -0,0 +1,60 @@
+        echo "Select the correct ending date";
+    } else if (empty($imageFile)) {
+        echo "Select an image";
+    } else if (strlen($imageFile) > 10000) {
```
This validation is wrong. Implement the proper validation to check the image `file size`.

dulanjayabhanu commented on loginPrcess.php:
```diff
@@ -26,6 +26,11 @@
             $loginData2 = $loginResultSet2->fetch_assoc();
+
+            $user = new stdClass();
+            $user->id = $loginData2["id"];
```
You can assign the `user` object to the `user session` associative array. Its very useful when you have to access user related data. And you can reduce additional search processes.

dulanjayabhanu commented on updatePromotionProcess.php:
```diff
@@ -0,0 +1,63 @@
+    date_default_timezone_set('Asia/Colombo');
+    $currentDate = date('Y-m-d H:i:s');
+
+    if (isset($_FILES["imagePath"])) {
```
Put that code fragment after the `UPDATE` process. In this case, there is some possibility that an `error` will occur during the update process. So the new data will **not be saved** to the database, but the image will **already be saved** to the file path. That's not appropriate...

dulanjayabhanu reviewed (approved):
I give my approval to this PR. Also, there are some minor issue here...fix that.

DilushaLakshan committed:
changed every process file and changed every client, staff side front-end files.

dulanjayabhanu commented on addCategoryProcess.php:
```diff
@@ -0,0 +1,22 @@
+            $categoryName = $_POST["categoryName"];
+
+            // insert the data
+            Database::insertUpdateDelete("INSERT INTO `destination_category` (`name`) VALUES ('" . $categoryName . "')");
```
In this approach, there is some possibility of adding duplicate data. First check whether the category name has already been added or not. After validation, you can enter a new category name.

dulanjayabhanu commented on addDestinationProcess.php:
```diff
@@ -1,93 +1,117 @@
-        for ($y = 0; $y < sizeof($categories); $y++) {
-            Database::insertUpdateDelete("INSERT INTO `destination_has_destination_category` (`destination_id`, `destination_category_id`) VALUES ('" . $desData['id'] . "', '" . $categories[$y] . "')");
+// Insert into the destination table
+Database::insertUpdateDelete("INSERT INTO `destination` (`name`, `description`) VALUES ('" . $name . "', '" . $description . "')");
```
First, check the destination is already exists or not.
